src/utils.ts

```diff
@@ -32,3 +32,22 @@ export function mergePaths(a: string, b: string) {
   if (!b.startsWith("/")) return a + "/" + b;
   return a + b;
 }
+
+/**
+ * Converts a file path to a valid JS export name.
+ *
+ * @example
+ * ```ts
+ * pathToExportName("/islands/foo.tsx");     // "foo"
+ * pathToExportName("/islands/foo.v2.tsx");  // "foo_v2"
+ * pathToExportName("/islands/nav-bar.tsx"); // "nav_bar"
+ * pathToExportName("/islands/_.$bar.tsx");  // "_$bar"
+ * pathToExportName("/islands/1.hello.tsx"); // "_hello"
+ * ```
+ */
+export function pathToExportName(filePath: string): string {
+  const name = path.basename(filePath, path.extname(filePath));
+  // Regex for valid JS identifier characters
+  const regex = /^[^a-z_$]|[^a-z0-9_$]/gi;
+  return name.replaceAll(regex, "_").replaceAll(/_{2,}/g, "_");
+}
```

Comment:
```suggestion

const INVALID_JS_IDENTIFIER = /^[^a-z_$]|[^a-z0-9_$]/gi;
const DOUBLE_UNDERSCORE = /_{2,}/g

/**
 * Converts a file path to a valid JS export name.
 *
 * @example
 * ```ts
 * pathToExportName("/islands/foo.tsx");     // "foo"
 * pathToExportName("/islands/foo.v2.tsx");  // "foo_v2"
 * pathToExportName("/islands/nav-bar.tsx"); // "nav_bar"
 * pathToExportName("/islands/_.$bar.tsx");  // "_$bar"
 * pathToExportName("/islands/1.hello.tsx"); // "_hello"
 * ```
 */
export function pathToExportName(filePath: string): string {
  return path.basename(filePath, path.extname(filePath))
    .replaceAll(INVALID_JS_IDENTIFIER, "_")
    .replaceAll(DOUBLE_UNDERSCORE, "_");
}
```